hadoop-yarn-project/hadoop-yarn/hadoop-yarn-client/src/main/java/org/apache/hadoop/yarn/client/api/impl/AHSv2ClientImpl.java

```diff
@@ -137,9 +137,8 @@ public List<ContainerReport> getContainers(ApplicationAttemptId
     ApplicationId appId = applicationAttemptId.getApplicationId();
     ApplicationReport appReport = getApplicationReport(appId);
     Map<String, String> filters = new HashMap<>();
-    filters.put("infofilters", "SYSTEM_INFO_PARENT_ENTITY eq {\"id\":\"" +
-        applicationAttemptId.toString() +
-        "\",\"type\":\"YARN_APPLICATION_ATTEMPT\"}");
+    filters.put("infofilters", "SYSTEM_INFO_PARENT_ENTITY eq "
+        + "{\"type\":\"YARN_APPLICATION_ATTEMPT\",\"id\":\"" + applicationAttemptId + "\"}");
```

Comment:
Is this completely relying on the toString representation of ApplicationAttemptId?
It's not a good practice to do so, if I understand it correctly.

Reply:
It is, and it was before the change, as this basically puts together a GET query, where it's expected to get the applicationAttempt string form. Also it's not likely that the ApplicationAttemptId will be changed because it's marked public and stable, and changing this means that the applicationAttempt format will change everywhere starting from the cli tools all the way to logging.

Reply:
Oh yeah, you are right :) Thanks for explaining.